capella_git_hooks/fix_links.py

```diff
@@ -43,6 +56,46 @@ def get_unmodified_tracked_files() -> list[str]:
         return []
 
 
+def search_and_replace(
+    pattern: re.Pattern, value: str, replacement: str
+) -> str | None:
```

Comment:
This will never return None, as far as I can tell.

```suggestion
def search_and_replace(
    pattern: re.Pattern, value: str, replacement: str
) -> str:
```

(But please do correct me if I'm wrong :) )